generator/views/view.py

```diff
@@ -104,26 +104,31 @@ def to_lookml(self, bq_client, v1_name: Optional[str]) -> Dict[str, Any]:
 
     def get_client_id(self, dimensions: List[dict], table: str) -> Optional[str]:
         """Return the first field that looks like a client identifier."""
-        client_id_fields = [
-            d["name"]
-            for d in dimensions
-            if d["name"] in {"client_id", "client_info__client_id", "context_id"}
-        ]
-        if not client_id_fields:
-            # Some pings purposely disinclude client_ids, e.g. firefox installer
-            return None
-        if len(client_id_fields) > 1:
-            raise ClickException(f"Duplicate client_id dimension in {table!r}")
-        return client_id_fields[0]
+        client_id_fields = self.select_dimension(
+            {"client_id", "client_info__client_id", "context_id"},
+            dimensions,
+            table,
+        )
+        return client_id_fields["name"] if client_id_fields else None
 
     def get_document_id(self, dimensions: List[dict], table: str) -> Optional[str]:
         """Return the first field that looks like a document_id."""
-        document_id_fields = [
-            d["name"] for d in dimensions if d["name"] in {"document_id"}
-        ]
-        if not document_id_fields:
+        document_id = self.select_dimension("document_id", dimensions, table)
+        return document_id["name"] if document_id else None
+
+    def select_dimension(
+        self,
+        dimension_names: str | Set[str],
+        dimensions: List[dict],
+        table: str,
+    ) -> Optional[dict[str, str]]:
+        """Return the first field that matches dimension name."""
+        if not isinstance(dimension_names, set):
+            dimension_names = {dimension_names}
+        selected = [d for d in dimensions if d["name"] in dimension_names]
+        if not selected:
             # Some pings purposely disinclude client_ids, e.g. firefox installer
             return None
-        if len(document_id_fields) > 1:
-            raise ClickException(f"Duplicate document_id dimension in {table!r}")
-        return document_id_fields[0]
+        if len(selected) > 1:
+            raise ClickException(f"Duplicate {dimension_names} dimension in {table!r}")
+        return selected[0]
```

Comment:
If you're looking for multiple dimension names, saying there's a duplicate if more than one of those dimensions is present doesn't seem accurate. I also think it'd be good to allow for callers to pass a prioritized list of dimensions, checking for those in the order the caller specified (implying the use of a list or other sequence rather than an unordered set), and returning the first one found.

So perhaps something like this instead:
```python
    def select_dimension(
        self,
        dimension_names: str | List[str],
        dimensions: List[dict],
        table: str,
    ) -> Optional[dict[str, str]]:
        """Return the first field that matches dimension name."""
        if isinstance(dimension_names, str):
            dimension_names = [dimension_names]
        for dimension_name in dimension_names:
            selected = [d for d in dimensions if d["name"] == dimension_name]
            if selected:
                if len(selected) > 1:
                    raise ClickException(f"Duplicate {dimension_name} dimension in {table!r}")
                return selected[0]
        return None
```

Reply:
Ok so I started to refactor this and it seems like we want the old behavior which is "Select the first item in the set, throw if more than one item in the set is selected"

Reply:
OK, I now see how you're preserving the previous behavior of `get_client_id()`.